code/modules/events/camerafail.dm

```diff
@@ -0,0 +1,51 @@
+/datum/round_event_control/camerafail
+	name = "Major Camera Failure"
+	typepath = /datum/round_event/camfail
+	weight = 50
+	max_occurrences = 100
+	earliest_start = 20
+	alertadmins = 1
+
+/datum/round_event/camfail
+	startWhen		= 0
+	endWhen			= 2
+	announceWhen	= 1
+	var/maxfailures = 50
+
+/datum/round_event/camfail/announce()
+	priority_announce("Unwarranted recording software detected in [station_name()]'s camera network, remotely disabling affected cameras","Camera Shutdown Alert")
+
+
+/datum/round_event/camfail/start()
+	var/disabledcount = 0
+	for( var/obj/machinery/camera/C in world)
```

Comment:
Because it's pretty much universally known that ```in world``` is hitler.
There's either a ```cameras``` list, or cameras are in the ```machines``` list, use the correct list instead of ```world```